### DISCUSSION TITLE
How to Extend and Migrate Product Schema in Medusa.js 2?

### DISCUSSION THREAD

**Discussion by soufianetomase** [medusajs/medusa#9417] [Q&A]
Hello everyone,

I'm currently working on a project using Medusa.js 2, and I need to update the structure of the existing Product entity to include some additional fields. In Medusa.js 1, I would typically handle this with migrations, but I'm unsure how to approach this in version 2.

Could someone provide guidance or examples on how to:

Extend the Product entity to add new fields.
Create and apply migrations for these changes in Medusa.js 2.
Any advice or resources would be greatly appreciated!

Thank you!

**Comment by fxmb** [comment]
Having the same issue. I posted it here: https://github.com/medusajs/medusa/discussions/9484

It's incredibly blocking that the docs seem to be wrong here and no one is responding. On discord it seems many devs have issues around this and find the new process incredibly cumbersome / not working.
